rust/operator-binary/src/discovery.rs

```diff
@@ -121,6 +124,7 @@ fn build_discovery_configmap(
                     hive,
                     APP_NAME,
                     hive_version(hive).unwrap(),
+                    RESOURCE_MANAGER_HIVE_CONTROLLER,
```

Comment:
The controller name must be provided in a parameter. Do not use the global constant here! If another controller will be added in the future which also uses discovery configmaps then this line must be changed otherwise there will be a bug which is really hard to detect. For instance, the kafka-operator builds discovery maps in two distinct controllers.

Reply:
I discussed with razvan, and the discovery CM does not need to be a cluster resource. It will be deleted with the cluster. What do you think?